### PR TITLE
Fix state-info icon color and convert to TypeScript/LitElement

### PR DESCRIPTION
**Pull request by iantrich**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #https://github.com/home-assistant/frontend/issues/6636
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/entity/state-info.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  @property({ type: Boolean, reflect: true }) public rtl = false;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    return html` <state-badge
+        .stateObj=${this.stateObj}
+        .stateColor=${true}
```
Comment (iantrich) [inline src/components/entity/state-info.ts:38]: we were not passing true for `stateColor` which is why the icon was not being colored

File: src/components/entity/state-info.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { HassEntity } from "home-assistant-js-websocket";
+
+import { computeStateName } from "../../common/entity/compute_state_name";
+import { computeRTL } from "../../common/util/compute_rtl";
+import { HomeAssistant } from "../../types";
```
Comment (zsarnett) [inline src/components/entity/state-info.ts:15]:
```suggestion
import type { HomeAssistant } from "../../types";
```

File: src/components/entity/state-info.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { HassEntity } from "home-assistant-js-websocket";
```
Comment (zsarnett) [inline src/components/entity/state-info.ts:11]:
```suggestion
import type { HassEntity } from "home-assistant-js-websocket";
```

File: src/components/entity/state-info.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+  // property used only in css
+  @property({ type: Boolean, reflect: true }) public rtl = false;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    return html` <state-badge
```
Comment (zsarnett) [inline src/components/entity/state-info.ts:36]:
```suggestion
    return html`<state-badge
```

File: src/components/entity/state-info.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        .stateObj=${this.stateObj}
+        .stateColor=${true}
+      ></state-badge>
+      <div class="info">
+        <div class="name" .inDialog=${this.inDialog}>
+          ${computeStateName(this.stateObj)}
+        </div>
+        ${this.inDialog
+          ? html` <div class="time-ago">
```
Comment (zsarnett) [inline src/components/entity/state-info.ts:45]:
```suggestion
          ? html`<div class="time-ago">
```

File: src/components/entity/state-info.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+              <ha-relative-time
+                id="last_changed"
+                .hass=${this.hass}
+                .datetime=${this.stateObj.last_changed}
+              ></ha-relative-time>
+              <paper-tooltip animation-delay="0" for="last_changed">
+                <div>
```
Comment (zsarnett) [inline src/components/entity/state-info.ts:52]: Extra div?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**